diff --git a/migen/build/xilinx/vivado.py b/migen/build/xilinx/vivado.py
index 88b6949..eb7646d 100644
--- a/migen/build/xilinx/vivado.py
+++ b/migen/build/xilinx/vivado.py
@@ -84,6 +84,7 @@ class XilinxVivadoToolchain:
         self.bitstream_commands = []
         self.additional_commands = []
         self.pre_synthesis_commands = []
+        self.post_synthesis_commands = []
         self.with_phys_opt = False
         self.clocks = dict()
         self.false_paths = set()
@@ -115,6 +116,7 @@ class XilinxVivadoToolchain:
         tcl.append("read_xdc {}.xdc".format(build_name))
         tcl.extend(c.format(build_name=build_name) for c in self.pre_synthesis_commands)
         tcl.append("synth_design -top {} -part {}".format(build_name, platform.device))
+        tcl.extend(self.post_synthesis_commands)
         tcl.append("report_timing_summary -file {}_timing_synth.rpt".format(build_name))
         tcl.append("report_utilization -hierarchical -file {}_utilization_hierarchical_synth.rpt".format(build_name))
         tcl.append("report_utilization -file {}_utilization_synth.rpt".format(build_name))
